src/avram/needy_initializer.cr

```diff
@@ -0,0 +1,102 @@
+module Avram::NeedyInitializer
```

Comment:
This module covers everything the `needy_initializer_with_save_methods` does, but without the save methods since those are related to SaveOperation specifically. I made a separate one because I had a lot of trouble with the compiler trying to tell "if it's SaveOp, then call these macros, else skip them". It was easier to just break them apart.